WooCommerce/Classes/ViewRelated/Products/Filters/FilterListViewController.swift

```diff
@@ -0,0 +1,159 @@
+
+    init(command: Command) {
+        self.command = command
+        super.init(nibName: "FilterListViewController", bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    deinit {
+        observationToken?.cancel()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureNavigation()
+        configureMainView()
+        configureClearAllActionVisibility()
+        configureChildNavigationController()
+        configureFilterActionContainerView()
+        observeShouldReloadUI()
+    }
+
+    // MARK: - Navigation
+    //
+    @objc func filterActionButtonTapped() {
+        command.onFilterActionTapped()
+    }
+
+    @objc func dismissButtonTapped() {
+        command.onDismissActionTapped()
+    }
+
+    @objc func clearAllButtonTapped() {
+        command.onClearAllActionTapped()
+    }
+}
+
+// MARK: - View Configuration
+//
+private extension FilterListViewController {
+    func configureNavigation() {
+        let dismissButtonTitle = NSLocalizedString("Dismiss", comment: "Button title for dismissing filtering a list.")
+        listSelector.navigationItem.leftBarButtonItem = UIBarButtonItem(title: dismissButtonTitle,
+                                                                        style: .plain,
+                                                                        target: self,
+                                                                        action: #selector(dismissButtonTapped))
+
+        let clearAllButtonTitle = NSLocalizedString("Clear all", comment: "Button title for clearing all filters for the list.")
+        clearAllBarButtonItem = UIBarButtonItem(title: clearAllButtonTitle, style: .plain, target: self, action: #selector(clearAllButtonTapped))
+
+        // Disables interactive dismiss action so that we can prompt the discard changes alert.
+        if #available(iOS 13.0, *) {
+            isModalInPresentation = true
+        }
+
+        listSelector.removeNavigationBackBarButtonText()
+    }
+
+    func configureMainView() {
+        view.backgroundColor = .listBackground
+    }
+
+    func configureClearAllActionVisibility() {
+        // Until we have a `PassthroughSubject`/`BehaviorSubject`, we have to initialize the Clear All CTA visibility manually.
+        listSelector.navigationItem.rightBarButtonItem = command.isClearAllActionVisible ? clearAllBarButtonItem: nil
+    }
+
+    func configureChildNavigationController() {
+        let navigationController = UINavigationController(rootViewController: listSelector)
+        addChild(navigationController)
+        navigationControllerContainerView.addSubview(navigationController.view)
+        navigationController.didMove(toParent: self)
+
+        navigationController.view.translatesAutoresizingMaskIntoConstraints = false
+        navigationControllerContainerView.pinSubviewToAllEdges(navigationController.view)
```

Comment:
thanks for the suggestion! since we want to show the bottom button container view fixed and only the top navigates when the user taps on a filter row, I think embedding a navigation controller might be the only way. otherwise, the view controller that's pushed wouldn't have the bottom button container. I think it's inevitable sometimes to have a particular way of presenting a view controller, like if it has navigation bar actions then we have to present it in a navigation controller and like in this case it has to be presented modally. since G signed off on the screenshots (including the landscape ones), does it sound okay to you to have the containers and landscape mode as it is for now? in the future, if we prefer not to embed a navigation controller, we can create another container view controller with the 2 containers in a vertical stack view and have `FilterListViewController` and a new view controller like `FilterValueListViewController` subclass it (also mentioned in https://github.com/woocommerce/woocommerce-ios/pull/2204#issuecomment-621830814)